components/molecules/videoPlayer/index.js

```diff
@@ -0,0 +1,94 @@
+import { useRef, useState } from 'react'
+import styled, { css } from 'styled-components'
+import { PlayCircleIcon, fullScreen, FullScreenIcon, ResizeDetector } from 'components'
+import { string } from 'prop-types'
+
+export const VideoPlayer = ({ src, poster, className }) => {
+  const [isPlaying, setIsPlaying] = useState(false)
+  const [width, setWidth] = useState(0)
+  const videoRef = useRef()
+  const played = useRef(false)
+
+  const handlePlayPause = () => {
+    isPlaying === false ? videoRef.current.play() : videoRef.current.pause()
+    setIsPlaying(state => !state)
+  }
+
+  const handlePaused = () => {
+    setIsPlaying(false)
+  }
+  const handlePlayed = () => (played.current = true)
+
+  const handleFullScreen = () => {
+    videoRef.current.play()
+    setIsPlaying(true)
+    fullScreen.requestFullscreen(videoRef.current)
+  }
+
+  const handleResize = ({ width }) => {
+    if (videoRef.current) {
+      const width = videoRef.current.clientWidth
+      setWidth(width)
+    }
+  }
+  const iconSize = width ? width / 6 : 40
+  return (
+    <VideoPlayerWrapper className={className}>
+      <ResizeDetector onResize={handleResize} />
+      <Overlay show={!isPlaying} poster={played.current ? null : poster} gap={iconSize / 3}>
+        <PlayCircleIcon size={iconSize} hoverColour onClick={handlePlayPause} />
+        <FullScreenIcon size={iconSize} hoverColour onClick={handleFullScreen} />
+      </Overlay>
+      <Video ref={videoRef} controls onPause={handlePaused} onPlay={handlePlayed}>
+        <source src={src} type='video/mp4' />
```

Comment:
Are there any other important attributes for `video` tag that should be passed to it? Should we spread rest of props on that? 
Please also make `type` attribute of `<source />` to read from props.